addons/web/static/src/core/utils/numbers.js

```diff
@@ -16,16 +16,23 @@ export function clamp(num, min, max) {
 
 /**
  * A function to create flexibly-numbered lists of integers, handy for each and map loops.
- * step defaults to 1.
  * Returns a list of integers from start (inclusive) to stop (exclusive), incremented (or decremented) by step.
- * @param {number} start default 0
- * @param {number} stop
- * @param {number} step default 1
+ *
+ * @param {number} start
+ * @param {number} [stop] if not provided, start is used as stop and start is set to 0.
+ * @param {number} [step] default to 1 if start < stop, -1 otherwise.
  * @returns {number[]}
  */
-export function range(start, stop, step = 1) {
+export function range(start, stop, step) {
     const array = [];
-    const nsteps = Math.floor((stop - start) / step);
```

Comment:
Note: `floor` was wrong, should be `ceil` (see test `expect(range(0, 34, 5)).toEqual([0, 5, 10, 15, 20, 25, 30]);` -> last 30 was not present)